packages/attestation-service/src/requestHandlers/get_attestation.ts

```diff
@@ -87,10 +89,24 @@ class GetAttestationRequestHandler {
 }
 
 export async function handleGetAttestationRequest(
-  _req: express.Request,
+  req: express.Request,
   res: express.Response,
   getRequest: GetAttestationRequest
 ) {
+  const {
+    headers: { authentication },
+  } = req
+  if (!authentication) {
+    respondWithError(res, 401, 'Missing authentication')
+    return
+  }
+
+  const typedData = buildSecurityCodeTypedData(getRequest.account)
+  if (!verifyEIP712TypedDataSigner(typedData, authentication as string, getRequest.account)) {
```

Comment:
This is gonna be tricky when the account is not an EOA but a SC/MTW. Thus I think you'll be wanting to check against the walletAddress or DEK here as well